libbpf-rs/src/map.rs

```diff
@@ -731,3 +734,117 @@ impl<'a> Iterator for MapKeyIter<'a> {
         }
     }
 }
+/// Represent a wrapper `bpf_map_info` struct pointer
+/// Provide ability to retrive the details of a certain map with its fd only
+#[derive(Debug)]
+pub struct MapInfo {
+    info: bpf_map_info,
+}
+
+impl MapInfo {
+    /// Create a MapInfo struct from a fd.
+    pub fn new(fd: i32) -> Result<Self> {
+        let mut map_info = unsafe { std::mem::zeroed::<bpf_map_info>() };
+        let mut size = std::mem::size_of_val(&map_info) as u32;
+        let ret = unsafe {
+            bpf_obj_get_info_by_fd(
+                fd,
+                &mut map_info as *mut bpf_map_info as *mut c_void,
+                &mut size as *mut u32,
+            )
+        };
+        if ret < 0 {
+            return Err(Error::System(errno::errno()));
+        }
```

Comment:
Please use `parse_ret_i32` helper.

Reply:
This function returns `Result<Self>`, not `Result<i32>`or `Result<usize>`. May be add a new `parse_ret_XXX` function?